packages/shared/components/atoms/BalanceSummarySection.svelte

```diff
@@ -1,72 +1,57 @@
+    $: convertedAmount = formatCurrency(getMarketAmountFromAssetValue(amount, baseCoin))
 </script>
 
-<div class="flex justify-between">
-    <div class={title ? 'flex flex-col space-y-0.5' : ''}>
-        <Text
-            color={PRIMARY_COLOR}
-            darkColor={PRIMARY_DARK_COLOR}
-            fontSize={PRIMARY_FONT_SIZE}
-            fontWeight={totalRow ? FontWeight.semibold : PRIMARY_FONT_WEIGHT}
-            lineHeight={PRIMARY_LINE_HEIGHT}
-        >
-            {title}
-        </Text>
-        {#if subtitle}
-            <Text
-                color={SECONDARY_COLOR}
-                darkColor={SECONDARY_DARK_COLOR}
-                fontSize={SECONDARY_FONT_SIZE}
-                fontWeight={SECONDARY_FONT_WEIGHT}
-                lineHeight={SECONDARY_LINE_HEIGHT}
-            >
-                {subtitle}
-            </Text>
+<div class="flex flex-col space-y-8">
+    <div
+        class="w-full flex flex-row flex-grow justify-between space-x-2 {hasChildren ? 'cursor-pointer ' : ''}"
+        on:click={() => (expanded = !expanded)}
+        on:keydown={() => {}}
+    >
+        {#if hasChildren}
+            <Icon
+                icon={expanded ? IconEnum.ChevronUp : IconEnum.ChevronDown}
+                width="12"
+                height="12"
+                classes="dark:text-white mt-1"
+            />
         {/if}
-    </div>
-    <div class={formattedAmount ? 'flex flex-col items-end space-y-0.5' : ''}>
-        <Text
-            color={PRIMARY_COLOR}
-            darkColor={PRIMARY_DARK_COLOR}
-            fontSize={PRIMARY_FONT_SIZE}
-            fontWeight={totalRow ? FontWeight.semibold : PRIMARY_FONT_WEIGHT}
-            lineHeight={PRIMARY_LINE_HEIGHT}
-        >
-            {formattedAmount}
-        </Text>
-        <Text
-            color={SECONDARY_COLOR}
-            darkColor={SECONDARY_DARK_COLOR}
-            fontSize={SECONDARY_FONT_SIZE}
-            fontWeight={SECONDARY_FONT_WEIGHT}
-            lineHeight={SECONDARY_LINE_HEIGHT}
-        >
+        <BalanceSummaryRow
+            title={titleKey ? localize(`popups.balanceBreakdown.${titleKey}.title`) : ''}
+            subtitle={subtitleKey ? localize(`popups.balanceBreakdown.${subtitleKey}.subtitle`) : ''}
+            amount={formattedAmount}
             {convertedAmount}
-        </Text>
+            {bold}
+        />
     </div>
+    {#if expanded}
+        {#each Object.keys(subBreakdown ?? {}) as breakdownKey}
+            <BalanceSummaryRow
+                title={localize(`popups.balanceBreakdown.${breakdownKey}.title`)}
+                subtitle={localize(`popups.balanceBreakdown.${breakdownKey}.subtitle`)}
+                amount={formatTokenAmountBestMatch(subBreakdown[breakdownKey].amount, baseCoin.metadata)}
+                convertedAmount={formatCurrency(
+                    getMarketAmountFromAssetValue(subBreakdown[breakdownKey].amount, baseCoin)
+                )}
```

Comment:
You use the same logic  on l.18-19, maybe it's worth refactoring? I'm not sure about this one, so feel free to resolve if you think it's overkill.